packages/magiclink-auth-service-api/azure-magic-link-web-app-build.yml

```diff
@@ -0,0 +1,147 @@
+# Docker
+# Build and push an image to Azure Container Registry
+# https://docs.microsoft.com/azure/devops/pipelines/languages/docker
+
+trigger:
+  branches:
+    include:
+      - master
+  paths:
+    include:
+      - packages/magiclink-auth-service-api
+
+resources:
+  - repo: self
+
+name: '0.0.0' # set dynamically below in a task
+
+variables:
+  # VERSION INFO ----------------------------------------------------------
+  version.MajorMinor: '1.0' # Manually adjust the version number as needed for semantic versioning. Revision is auto-incremented.
+  version.Revision: $[counter(variables['version.MajorMinor'], 0)]
+  versionNumber: '$(version.MajorMinor).$(version.Revision)'
+  # -----------------------------------------------------------------------
+
+  # Container registry service connection established during pipeline creation
+  lpaQuestionnaireWebAppDockerfilePath: '$(Build.SourcesDirectory)/packages/magiclink-auth-service-api/Dockerfile'
+  lpaQuestionnaireWebAppContext: '$(Build.SourcesDirectory)/packages/magiclink-auth-service-api'
+  imageRepository: 'magiclink-auth-service-api'
+  containerRegistry: 'pinscommonukscontainers3887default.azurecr.io'
+  dockerRegistryServiceConnection: 'appeals-service-api-registry-connection' ## this needs renamed
+  # Agent VM image name
+  vmImageName: 'ubuntu-latest'
+
+stages:
+  - stage: Increment_Version
+    displayName: Increment Version Number
+    jobs:
+      - job: Set_Version
+        displayName: Set Version Number
+        pool:
+          vmImage: $(vmImageName)
+        steps:
+          - task: PowerShell@2
+            displayName: Set the name of the build (i.e. the Build.BuildNumber)
+            inputs:
+              targetType: 'inline'
+              script: |
+                [string] $buildName = "$(versionNumber)_$(Build.BuildId)"
+                Write-Host "Setting the name of the build to '$buildName'."
+                Write-Host "##vso[build.updatebuildnumber]$buildName"
+
+  - stage: Build_and_Push_Magic_Link_API
+    displayName: Build, Test and Push Magic Link API
+    jobs:
+      - job: Build_Test_Push_Magic_Link_API
+        displayName: Build Magic_Link_API
+        pool:
+          vmImage: $(vmImageName)
+        steps:
+          - task: Docker@2
+            displayName: Build and Test Image
+            inputs:
+              command: build
+              repository: $(imageRepository)
+              dockerfile: $(lpaQuestionnaireWebAppDockerfilePath)
+              containerRegistry: $(dockerRegistryServiceConnection)
+              buildContext: $(lpaQuestionnaireWebAppContext)
+              arguments: >
+                --build-arg APP_NAME=lpa-questionnaire-web-app
+                --build-arg BUILD_ID=$(Build.BuildNumber)
+                --build-arg VERSION=$(versionNumber)
+              tags: |
+                $(versionNumber)
+                lpa-questionnaire-web-app
```

Comment:
tag needs to be unique otherwise will deploy lpa-questionnaire to magic link infra